src/CMakeLists.txt

```diff
@@ -1,5 +1,165 @@
 ##
-# Lists of source files. Keep ALL lists sorted in alphabetical order.
+# Set up additional backend options for Halide
+##
+option(TARGET_OPENCL "Include OpenCL-C target" ON)
+option(TARGET_OPENGL "Include OpenGL/GLSL target" ON)
+option(TARGET_METAL "Include Metal target" ON)
+option(TARGET_D3D12COMPUTE "Include Direct3D 12 Compute target" ON)
+
+##
+# List of optional targets
+##
+
+set(OPTIONAL_TARGETS
```

Comment:
Where possible, I would like to enforce the convention that `ALL_CAPS` variables are reserved for _cache_ variables and function arguments. Normal variables like `optional_targets` here should be `lower_snake_case`.